client/blocks/reader-post-card/index.jsx

```diff
@@ -27,6 +27,79 @@ import { getDiscoverBlogName,
 } from 'reader/discover/helper';
 import DiscoverFollowButton from 'reader/discover/follow-button';
 
+function trackable( TrackedComponent ) {
+	return class Trackable extends React.Component {
+		constructor( props ) {
+			super( props );
+			this.state = { isOnScreen: false };
+		}
+
+		defaultProps = {
+			onAppear: noop,
+			onLeave: noop
+		}
+
+		componentDidMount() {
+			window.addEventListener( 'scroll', this.checkOnScreen );
+			window.addEventListener( 'resize', this.checkOnScreen );
+			this.checkOnScreen();
+		}
+
+		componentWillUnmount() {
+			window.removeEventListener( 'scroll', this.checkOnScreen );
+			window.removeEventListener( 'resize', this.checkOnScreen );
+		}
+
+		checkOnScreen = throttle( () => {
+			if ( ! this.nodeRef ) {
+				return;
+			}
+			const rect = this.nodeRef.getBoundingClientRect();
+			const html = document.documentElement;
+			const windowHeight = window.innerHeight || html.clientHeight;
+
+			function entirelyOnScreen() {
```

Comment:
yeah, you can be entirely on screen and not trip the partially on screen if the node is smaller than 1/4 of the screen.